docker-compose.yml

```diff
@@ -57,6 +59,11 @@ services:
     environment:
       - MINIO_ROOT_USER=admin
       - MINIO_ROOT_PASSWORD=password
+      - MINIO_DOMAIN=minio
```

Comment:
this is required for Minio to enable virtual-host-style access to the bucket (which is the default in the new AWS SDK version

Reply:
all of the networking changes are basically only to make virtual-host-style access to the bucket properly work

Reply:
the alternative would be to configure path-style access (which is deprecated) similar to https://github.com/tabular-io/docker-spark-iceberg/pull/70